### DISCUSSION TITLE
Specify remote server kernel

### DISCUSSION THREAD

**Discussion by realratchet** [microsoft/vscode-jupyter#9142] [Questions and Answers] [answered]
Hello,

I am using code-server with Jupyter notebook plugin. It works good enough apart for some bugs in the plugin itself regarding external file updates confusing it. However, I have a critical issue with `jupyter.jupyterServerType` setting. There doesn't seem to be a Non-UI way of specifying the address and the kernel runtime and I need these settings to be already set when the code-server launches.

The settings seem to be persistent as it stays selected when restarting code-server & the tab. Yet it is not in User/Workspace settings so they seem to be cached somewhere. Can anyone fill me in where is this or is there a more sane way to do this.

**Comment by DonJayamanne** [comment, marked as answer]
> There doesn't seem to be a Non-UI way of specifying the address and the kernel runtime and I need these settings to be already set when the code-server launches.

Unfortunately we don't allow storing these in the settings as users could have sensitive information such as `tokens` used to authenticate against the Jupyter server.
We could support this provided you configure the jupyter server to allow anonymous connections (i.e. without any tokens, without passwords).
If this is an approach you're happy to go down, please feel free to file an issue as an enhancement.

**Reply by realratchet** [reply]
To be honest this is a workaround to the fact that `local` connection will always start a new kernel when refreshing the tab, so I'm fine with it as this will always be isolated anyway.
